universe/config.json

```diff
@@ -103,6 +103,52 @@
                     "type": "string",
                     "description": "Specify the integer UID for the Linux user, overriding service.user, when running the Spark Driver and Executor containers with the Docker Engine. Directly translates to docker run '--user' parameter. Note: This should typically be set to 99 when running as nobody (default) on RHEL/CentOS.",
                     "default": ""
+                },
+                "tls": {
+                    "type": "object",
+                    "description": "Spark TLS configuration.",
+                    "properties": {
+                        "enabled": {
+                            "description": "Enable TLS",
+                            "type": "boolean",
+                            "default": false
+                        },
+                        "protocol": {
+                            "description": "Protocol used for TLS",
+                            "type": "string",
+                            "default": ""
+                        },
+                        "keystore": {
```

Comment:
better use file-based secrets and treat keystore/keypassword/truststore/truststore_password as secret information

Reply:
+1 to that. Here's an example: https://github.com/mesosphere/dcos-commons/pull/3016/files#diff-56550f345e72e9ddaffbc774bb87a7a9R129